.github/workflows/test-windows.yml

```diff
@@ -0,0 +1,92 @@
+---
+name: Test Windows
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+
+      runner:
+        description: >
+          Describes the system this workflow should run on.
+        type: string
+        required: true
+
+      install-optionals:
+        description: >
+          Decides whether we install optyional dependencies
+        type: boolean
+        default: false
+jobs:
+  test-windows:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ inputs.runner }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ inputs.python-version }}
+        id: python-outputs
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - name: Cache stestr
+        uses: actions/cache@v4
+        with:
+          key: 'stestr | "${{ runner.os }}" | "${{ inputs.python-version }}" | "${{ github.run_id }}"'
+          restore-keys: |
+            stestr | "${{ runner.os }}" | "${{ inputs.python-version }}"
+            stestr | "${{ runner.os }}"
+            stestr
+          path: .stestr
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          python -m venv test-job
+          .\test-job\Scripts\activate
+          python -m pip install -U pip setuptools wheel
+          python -m pip install -U `
+            -c constraints.txt `
+            -r requirements.txt `
+            -r requirements-dev.txt `
+            -e .
+          pip check
```

Comment:
This probably isn't needed here because it's a single pip command. This was originally needed before pip added a new dependency solver to validate your weren't relying on implicit behavior that would break. It's good to run on two successive pip calls still to check that you didn't break a dependency from the first install with the second (because the depsolver is only invoked per session and not shared across multiple calls).